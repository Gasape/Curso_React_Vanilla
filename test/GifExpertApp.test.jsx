import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"


describe('Pruebas en <GifExpertApp />', () => {
    

    test('debe de hace match con el snapshot', () => { 

        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();

    });

    test('El formulario esta creando una nueva categoria', () => { 
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');

        const form = screen.getByRole('form');

        const inputValue = 'Bunny girl senpai';

        fireEvent.input( input, {target : { value: inputValue}});

        fireEvent.submit(form);

        expect( screen.getByText(inputValue).textContent ).toContain(inputValue);
        // screen.debug();

    })

    test('El formulario no esta creando una categoria que ya existe', () => { 
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');

        const form = screen.getByRole('form');

        const inputValue = 'Bunny girl senpai';

        fireEvent.input( input, {target : { value: inputValue}});

        fireEvent.submit(form);

        fireEvent.input( input, {target : { value: inputValue}});

        fireEvent.submit(form);

        expect( screen.getAllByText(inputValue).length ).toBe(1);
        // screen.debug();

    })
});