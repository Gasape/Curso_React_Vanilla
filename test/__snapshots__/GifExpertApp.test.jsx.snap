// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de hace match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2
    class=""
  >
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
